azurerm/resource_arm_app_service.go

```diff
@@ -202,7 +202,7 @@ func resourceArmAppServiceCreate(d *schema.ResourceData, meta interface{}) error
 	}
 
 	resGroup := d.Get("resource_group_name").(string)
-	location := azureRMNormalizeLocation(d.Get("location").(string))
+	location := azure.NormalizeLocation(d.Get("location"))
```

Comment:
`azure.NormalizeLocation()` takes an interface and does the cast itself

Reply:
in general we're moving away from having methods make assumptions in favour of them being explicit (e.g. in the case of expand functions passing in `[]interface{}` rather than casting within the method) - so I think it'd be worth doing the same here? the benefit of doing this is we'll get compile-time errors if this isn't valid?